File: php/features.php
<?php
declare(strict_types=1);

// Sotaproject SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SotaprojectFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SotaprojectBaseFeature();
            case "test":
                return new SotaprojectTestFeature();
            default:
                return new SotaprojectBaseFeature();
        }
    }
}

File: rb/features.rb
# Sotaproject SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SotaprojectFeatures
  def self.make_feature(name)
    case name
    when "base"
      SotaprojectBaseFeature.new
    when "test"
      SotaprojectTestFeature.new
    else
      SotaprojectBaseFeature.new
    end
  end
end
